ktor-client/ktor-client-features/ktor-client-json/ktor-client-moshi/jvm/src/io/ktor/client/features/json/MoshiSerializer.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.features.json
+
+import com.squareup.moshi.*
+import io.ktor.client.call.*
+import io.ktor.http.content.*
+import io.ktor.http.*
+import io.ktor.util.*
+import io.ktor.utils.io.core.*
+import okio.*
+import kotlin.reflect.*
+
+/**
+ * [JsonSerializer] using [Moshi] as backend.
+ */
+@ExperimentalStdlibApi
+class MoshiSerializer(block: Moshi.Builder.() -> Unit = {}) : JsonSerializer {
```

Comment:
Any reason to be inconsistent here by having a builder block passed to the client instead of a configured `Moshi` instance like the server implementation?

Reply:
Maybe a secondary constructor that accepts the builder block could allow it to be consistent and work with the block

```kotlin
constructor(block: Moshi.Builder.() -> Unit) : this(Moshi.Builder().apply(block).build())
```

Reply:
I just matched the Gson serializer as convention. I personally don't like this and would prefer it just accept a Moshi instance and let the user build it separately.

Reply:
A secondary constructor is what I did with [my implementation](https://github.com/plannigan/ktor-moshi/blob/dd3c55063ec598b71581d7db0d046c000dac4cfa/client/src/main/kotlin/com/hypercubetools/ktor/moshi/MoshiSerializer.kt#L13). It provides a bit more flexibility.

Reply:
Let's see what the ktor maintainers think